### ISSUE TITLE
Using shells.ghc from reflex-platform: fails because of the example

### ISSUE THREAD

**Issue by rubenmoor** [imalsogreg/servant-reflex#106]
Using `servant-reflex` in my `reflex-platform` project, everything works smoothly and I can use nix to build a package and deploy just fine.

However, for my IDE (haskell-language-server) I build my project, including frontend, with GHC instead of ghcjs. `reflex-platform` allows that just fine.

But since I use `servant-reflex` I get this error upon entering

```
$> nix-shell -A shells.ghc
Preprocessing executable 'example' for servant-reflex-0.3.6..
Building executable 'example' for servant-reflex-0.3.6..
[1 of 2] Compiling API              ( exec/API.hs, dist/build/example/example-tmp/API.o )
[2 of 2] Compiling Main             ( exec/Example.hs, dist/build/example/example-tmp/Main.o )

exec/Example.hs:34:8: error:
    • Couldn't match expected type ‘IO ()’
                  with actual type ‘jsaddle-0.9.7.0:Language.Javascript.JSaddle.Types.JSM
                                      ()’
    • In the expression:
        mainWidget
          $ do divClass "example-base" run
               divClass "example-multi" runMulti
      In an equation for ‘main’:
          main
            = mainWidget
                $ do divClass "example-base" run
                     divClass "example-multi" runMulti
   |
34 | main = mainWidget $ do
   |        ^^^^^^^^^^^^^^^...

builder for '/nix/store/m5pfw964cj104spnzgd81h347carmz3c-servant-reflex-0.3.5.drv' failed with exit code 1
cannot build derivation '/nix/store/3iga76ylnzmgik9k9adsidw6vagnhn5w-ghc-8.6.5-with-packages.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3iga76ylnzmgik9k9adsidw6vagnhn5w-ghc-8.6.5-with-packages.drv' failed
```

I don't know why the example gets built and fails, with GHC while ghcjs works fine.

**Comment by rubenmoor** [comment]
As a workaround, I disabled the example by overriding `configureFlage = [ "-fexample"]`:

    servant-reflex = (self.callPackage ./servant-reflex { }).overrideDerivation( _: { configureFlags = []; });

I am still curious, though, why the example wouldn't build.

**Comment by imalsogreg** [comment]
Thanks for reporting this! It seems I used a `mainWidget` in the example that only works under ghcjs. #107 fixes this and adds a test that the example compiles under ghc.

**Comment by imalsogreg** [comment]
Closed by #107
